src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines;
+        if (type == Bulldozer.class) {
+            machines = new BulldozerProducer().get();
+        } else if (type == Excavator.class) {
+            machines = new ExcavatorProducer().get();
+        } else if (type == Truck.class) {
+            machines = new TruckProducer().get();
+        } else {
+            List<Machine> emptyList = Collections.emptyList();
+            return emptyList;
+        }
```

Comment:
Keep if-else constructions simple. There's no need for an else clause after a return statement. You can simplify the code by removing the else keyword and unindenting the code that follows.